### DISCUSSION TITLE
virtualization.0016_replicate_interfaces migration fails with IntegrityError

### DISCUSSION THREAD

**Discussion by sschols** [netbox-community/netbox#7742] [Q&A]
### NetBox version

2.11.12

### Python version

3.8

### Steps to Reproduce

I'm upgrading a docker-based netbox install from 2.5.13 to 2.11.12 (as a intermediary step to 3.0.x), and I get the following migration error:
```
Running migrations:
  Applying virtualization.0016_replicate_interfaces...
    Replicating 22 VM interfaces...
Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "virtualization_vminterfa_virtual_machine_id_name_cfb9c423_uniq"
DETAIL:  Key (virtual_machine_id, name)=(5, Network adapter 1) already exists.
```

### Expected Behavior

The migration to go through smoothly and have a running netbox instance again.

### Observed Behavior

The error message above pops up, and netbox doesn't come up.

I found #5334 which seems to match the error, but wasn't followed up properly. 

We used to have a sync script from VMware to NetBox, maybe this caused this issue? We don't use NetBox for VMs anymore, so if there would be a way to remove all that from the database, and that would solve the issue, we'd be happy to do that.

**Comment by jeremystretch** [comment]
> Key (virtual_machine_id, name)=(5, Network adapter 1) already exists.

This error indicates that you have multiple interfaces belonging to the same VM with duplicate names. You'll need to delete the offending interfaces, then re-run the migrations.

**Reply by sschols** [reply]
Thanks for the reply. How would I do that? The application (in the docker container) doesn't seem to start without the migration being run.

**Reply by jeremystretch** [reply]
You'll need to remove it from the database somehow prior to the migrations being run. If you have a backup on the original deployment, it should be as simple as starting it up, finding the VM with the ID 5, and removing the duplicate interface(s). (Of course, there may be other duplicates to remove as well.)
